pkg/kubernetes/watch/watcher.go

```diff
@@ -198,6 +198,23 @@ func (w *Watcher[T]) run(ctx context.Context, listCallback func([]T) (bool, erro
 			return err
 		}
 		if ok, err := listCallback(items); err != nil {
+			// Check if the err is a "proper" API server error
+			if status, ok := err.(apierrors.APIStatus); ok || errors.As(err, &status) {
```

Comment:
Do you really need the type assertion? Couldn't this just be a test to `errors.As()`?

Reply:
I was wondering the same when taking over this pattern from [`k8s.io/apimachinery/pkg/api/errors`](https://github.com/kubernetes/kubernetes/blob/v1.25.3/staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go) where it's used quite frequently.

I just double-checked with the k8s git history. Seems this was [a recent performance improvement in Kubernetes](https://github.com/kubernetes/kubernetes/commit/13ebe3b454d0739ab1069cfbc37e26c1773874b1). Arguably this won't have measurable impact on k0s, so I'm happy to change that if you prefer.